internal/pkg/manifest/validate_test.go

```diff
@@ -48,6 +48,99 @@ func TestLoadBalancedWebService_validate(t *testing.T) {
 			},
 			wantedErrorMsgPrefix: `validate "image": `,
 		},
+		"error if fail to validate grace_period when specified in the additional listener rules of ALB": {
+			lbConfig: LoadBalancedWebService{
+				LoadBalancedWebServiceConfig: LoadBalancedWebServiceConfig{
+					ImageConfig: testImageConfig,
+					HTTPOrBool: HTTPOrBool{
+						HTTP: HTTP{
+							Main: RoutingRule{
+								Path: stringP("/"),
+								HealthCheck: HealthCheckArgsOrString{
+									Union: AdvancedToUnion[string](HTTPHealthCheckArgs{
+										Path:               aws.String("/testing"),
+										HealthyThreshold:   aws.Int64(5),
+										UnhealthyThreshold: aws.Int64(6),
+										Interval:           durationp(78 * time.Second),
+										Timeout:            durationp(9 * time.Second),
+									}),
+								},
+							},
+							AdditionalRoutingRules: []RoutingRule{
+								{
+									Path: stringP("/"),
+									HealthCheck: HealthCheckArgsOrString{
+										Union: AdvancedToUnion[string](HTTPHealthCheckArgs{
+											Path:               aws.String("/testing"),
+											HealthyThreshold:   aws.Int64(5),
+											UnhealthyThreshold: aws.Int64(6),
+											Interval:           durationp(78 * time.Second),
+											Timeout:            durationp(9 * time.Second),
+											GracePeriod:        durationp(9 * time.Second),
+										}),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantedErrorMsgPrefix: `validate "grace_period": cannot define "grace_period" in "http.additional_rules[0].healthcheck.grace_period"`,
```

Comment:
```suggestion
			wantedError: &errGracePeriodSpecifiedInAdditionalRule{0},
```

Reply:
The thing is if we write `wantedError: &errGracePeriodSpecifiedInAdditionalRule{0},` then we only get error as `cannot define "grace_period" in "http.additional_rules[0].healthcheck.grace_period"` without prefix as `validate "grace_period":` and our validate function expects the error with `validate "grace_period":`. So I think what we have right now will work better.

Reply:
if we write `wantedError: &errGracePeriodSpecifiedInAdditionalRule{0},` then we get error as `cannot define "grace_period" in "http.additional_rules[0].healthcheck.grace_period"` without prefix as `validate "grace_period":` which is expected from the validate function. So what we have right now works better in this case.

Reply:
hmmmmm in that case we can do 
```
fmt.Errorf(`validate "grace_period"`)
```

`wantedErrorPrefix` is used to validate error prefixes such as "validate xxx" and is not meant to validate that the errors are equals... and if we change the error string returned by `errGracePeriodSpecifiedInAdditionalRule`'s `Error()`, we will have to modify this test case again.